lib/rubygems/local_remote_options.rb

```diff
@@ -24,8 +24,10 @@ def accept_uri_http
         raise OptionParser::InvalidArgument, value
       end
 
-      unless ['http', 'https', 'file', 's3'].include?(uri.scheme)
-         raise OptionParser::InvalidArgument, value
+      allowed_schemes = ['http', 'https', 'file', 's3']
+      unless allowed_schemes.include?(uri.scheme)
+        msg =  "#{value}\nPreface URLs with one of #{allowed_schemes}"
+        raise OptionParser::InvalidArgument, msg
```

Comment:
I'm able to get back a custom error message if I don't use an instance of ```OptionParser::InvalidArgument```. I'll just throw a standard runtime error.